### PR TITLE
Improve performance of lib/statsd

### PR DESCRIPTION
**Pull request by Raynos**
This PR mostly optimizes the `clean()` functions in the
statsd integration.

r: @kriskowal @rf @shannili

### CODE REVIEW

File: node/lib/statsd.js [hunk 1]
```diff
@@ -36,53 +39,88 @@ function TChannelStatsd(channel, statsd) {
...
-        // TODO generalize
-        return str
-            .replace(/:/g, '-')
-            .replace(/\//g, '-')
-            .replace(/\./g, '-')
-            .replace(/{|}/g, '-');
+        return field;
     }
+
+    copy = LENGHT_ARRAYS[str.length];
```
Comment (kriskowal) [inline node/lib/statsd.js]: LENGTH

File: node/lib/statsd.js [hunk 2]
```diff
@@ -36,53 +39,88 @@ function TChannelStatsd(channel, statsd) {
...
+            char === '{' ||
+            char === '}'
+        ) {
+            copy[i] = '-';
+        } else {
+            copy[i] = char;
+        }
+    }
+
+    return copy.join('');
```
Comment (kriskowal) [inline node/lib/statsd.js]: I believe the prevailing wisdom is that string concatenation is faster than this.

**Comment by kriskowal** [comment]
:ok:

**Comment by ShanniLi** [comment]
lgtm

File: node/lib/statsd.js (hunk 2 above)
Comment (Raynos) [inline node/lib/statsd.js]: it's not; the flames were like "WHAT IS THIS CONCAT STRING DATA STRUCTURE"

Comment (Raynos) [inline node/lib/statsd.js]: wait I didn't try concatenation.

**Pull request merged**